atlasdb-api/src/main/java/com/palantir/atlasdb/spi/AtlasDbFactory.java

```diff
@@ -38,6 +38,13 @@
 
     String getType();
 
+    default MERGED_CONFIG createMergedKeyValueServiceConfig(
```

Comment:
Could we add a bit of docs saying how this function is expected to be used? In particular, values read off this merged config, even if the values originate from the runtime config, may not necessarily be respected for live reloading.